### DISCUSSION TITLE
How do I use multiplication in a memory address?

### DISCUSSION THREAD

**Discussion by 7x-hex-x7** [asmjit/asmjit#371] [Q&A] [answered]
I need to write the following on asmjit: 
`cmp dword ptr [ecx + eax * 4 + 0C], 02`

But so far i dont have found a way to use multiplication. I tried this:
`asm_01.cmp(dword_ptr(ecx, eax * 0x04, 0x0C), 0x02);`
but an error is displayed for this symbol: *

That is it should get the following bytes: 83 7C 81 0C 02

**Comment by petr-sneller** [comment, marked as answer]
Use shift `dword_ptr(ecx, eax, 2, 0x0C)` -> `ecx + eax << 2 + 0xC`

**Reply by 7x-hex-x7** [reply]
Oh, thank u, it is working!
